### PR TITLE
fix(lv_obj_style): initialise value_act with default values

### PR DESCRIPTION
**Pull request by douglas-healy**
### Description of the feature or fix
This change fixes a compiler warning indicating that `value_act` may be used uninitialised.
When declaring `value_act` initialise it with default values.

### CODE REVIEW

File: src/core/lv_obj_style.c [hunk 1]
```diff
@@ -224,7 +224,7 @@ void lv_obj_enable_style_refresh(bool en)
 
 lv_style_value_t lv_obj_get_style_prop(const lv_obj_t * obj, lv_part_t part, lv_style_prop_t prop)
 {
-    lv_style_value_t value_act;
+    lv_style_value_t value_act = {.num = 0, .ptr = NULL, .color.full = 0};
```
Comment (kisvegabor) [inline src/core/lv_obj_style.c]: It produces a warning. Maybe `lv_style_value_t value_act = { 0 }`

File: src/core/lv_obj_style.c [hunk 2]
```diff
@@ -224,7 +224,7 @@ void lv_obj_enable_style_refresh(bool en)
 
 lv_style_value_t lv_obj_get_style_prop(const lv_obj_t * obj, lv_part_t part, lv_style_prop_t prop)
 {
-    lv_style_value_t value_act;
+    lv_style_value_t value_act = { .num = 0 };
```
Comment (kisvegabor) [inline src/core/lv_obj_style.c]:
```suggestion
    lv_style_value_t value_act = { .ptr = NULL };
```

On 64 bit machines `num` is `int32_t` (4 bytes) but `ptr` is 8 bytes.

File: src/core/lv_obj_style.c (hunk 1 above)
Comment (douglas-healy) [reply src/core/lv_obj_style.c]: Hi @kisvegabor, I have pushed a new commit. I realised that `value_act` is a union and for some reason I was thinking it is a struct. I have now just initialised the `num` member to zero 😊

**Comment by kisvegabor** [comment]
I made a minor change, and all tests pass now. Merging, thank you!

**Comment by douglas-healy** [comment]
@kisvegabor, I agree with the change. Thank you for reviewing.

**Pull request merged**